doc/rst/meta/static/searchtools.js

```diff
@@ -0,0 +1,756 @@
+ *
+ * :copyright: Copyright 2007-2023 by the Sphinx team, see AUTHORS.
+ * :license: BSD, see LICENSE for details.
+ *
+ */
+"use strict";
+
+/**
+ * Simple result scoring code.
+ */
+if (typeof Scorer === "undefined") {
+  var Scorer = {
+    // Implement the following function to further tweak the score for each result
+    // The function takes a result array [docname, title, anchor, descr, score, filename]
+    // and returns the new score.
+    /*
+    score: result => {
+      const [docname, title, anchor, descr, score, filename] = result
+      return score
+    },
+    */
+
+    // query matches the full name of an object
+    objNameMatch: 11,
+    // or matches in the last dotted part of the object name
+    objPartialMatch: 6,
+    // Additive scores depending on the priority of the object
+    objPrio: {
+      0: 15, // used to be importantResults
+      1: 5, // used to be objectResults
+      2: -5, // used to be unimportantResults
+    },
+    //  Used when the priority is not in the mapping.
+    objPrioDefault: 0,
+
+    // query found in title
+    title: 15,
+    partialTitle: 7,
+    // query found in terms
+    term: 5,
+    partialTerm: 2,
+  };
+}
+
+const _removeChildren = (element) => {
+  while (element && element.lastChild) element.removeChild(element.lastChild);
+};
+
+/**
+ * See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+ */
+const _escapeRegExp = (string) =>
+  string.replace(/[.*+\-?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
+
+const _displayItem = (item, searchTerms, highlightTerms) => {
+  const docBuilder = DOCUMENTATION_OPTIONS.BUILDER;
+  const docFileSuffix = DOCUMENTATION_OPTIONS.FILE_SUFFIX;
+  const docLinkSuffix = DOCUMENTATION_OPTIONS.LINK_SUFFIX;
+  const showSearchSummary = DOCUMENTATION_OPTIONS.SHOW_SEARCH_SUMMARY;
+  const contentRoot = document.documentElement.dataset.content_root;
+
+  const [docName, title, anchor, descr, score, _filename, showTitle] = item;
+
+  let listItem = document.createElement("li");
+  let typestr = "";
+  if (docName.startsWith("language/spec/"))
+    typestr = "spec";
+  else if (docName.startsWith("primers/"))
+    typestr = "primer";
+  else if (docName.startsWith("examples/"))
+    typestr = "example";
+  else if (docName.startsWith("users-guide/"))
+    typestr = "users guide";
+  else if (docName.startsWith("usingchapel/man"))
+    typestr = "man page";
+  else if (docName.startsWith("technotes"))
+    typestr = "technical note";
+  else if (docName.startsWith("developer/compiler-internals/"))
+    typestr = "compiler internals";
+  else if (docName.startsWith("modules/"))
+    typestr = "standard library";
+
+  let useTitle = title;
+  if (showTitle) useTitle = showTitle;
+  let mydescr;
+  if (typestr || descr) {
+    mydescr = " (";
+    if (typestr != "") mydescr += typestr;
+    if (descr) {
+      if (typestr != "") mydescr += " ";
+      mydescr += descr;
+    }
+    mydescr += ")";
+  }
```

Comment:
We can make this a little more readable by using some template literals:
```js
const useTitle = showTitle || title;
let mydescr = "";

if (typestr || descr) {
  let typeDescription = '';

  // If typestr is truthy (i.e., it's a non-empty string), add it to the type description
  if (typestr) {
    typeDescription += typestr;

    // If descr is also truthy, add a space after typestr
    if (descr) {
      typeDescription += ' ';
    }
  }
  mydescr = ` (${typeDescription}${descr || ''})`;
}
```
It also removes a redundant check for `typestr` being non-empty and uses "truthy" javascript checks (which include checks against values like `null` and `undefined`, rather than just `!= ""`)